Documentation/Specification/frames.tex

```diff
@@ -200,8 +200,9 @@
 consists of additional arguments to be passed to the function.  The default
 function is \cl{default-frame-top-level}.
 
-\item \cl{:icon} \arg{pixmap} specifies a pixmap to be displayed by the host's
-window manager when the frame is iconified.
+\item \cl{:icon} \arg{pattern} specifies an array pattern to be
+displayed by the host's window manager to visually enhance or
+represent the frame, for example when it is iconified.
```

Comment:
> do we want to be able to specify also icon-name? _net_wm_icon_name in x11 terms. It is the string associated with the window when it is iconified (i.e shorter than a name - "firefox" vs "icons for top-level sheets and frames by scymtym * pull request * Mozilla Firefox") - i.e in alt+tab switching of windows.

I'm using Gnome Shell and it doesn't seem to use `_NET_WM_ICON_NAME` but instead something else for the short name (maybe something from the `.desktop` file? who knows).
![gnome-shell](https://user-images.githubusercontent.com/150189/98587836-00ad6080-22cb-11eb-98b5-4fa3ce9b2e46.png)
`xprop` for the Firefox window says:

```
WM_CLASS(STRING) = "Navigator", "Firefox-esr"
WM_ICON_NAME(STRING) = "Icons for top-level sheets and frames by scymtym · Pull Request #1098 · McCLIM/McCLIM - Mozilla Firefox"
_NET_WM_ICON_NAME(UTF8_STRING) = "Icons for top-level sheets and frames by scymtym · Pull Request #1098 · McCLIM/McCLIM - Mozilla Firefox"
WM_NAME(STRING) = "Icons for top-level sheets and frames by scymtym · Pull Request #1098 · McCLIM/McCLIM - Mozilla Firefox"
_NET_WM_NAME(UTF8_STRING) = "Icons for top-level sheets and frames by scymtym · Pull Request #1098 · McCLIM/McCLIM - Mozilla Firefox"
```
This is basically what we already do: we set all four properties to the "pretty name" of the top-level sheet.

There must, however, be an additional way to compute the name: the "frame and sheet name" demo gets the name "Frame-Sheet-Name-Test" in the Gnome Shell top bar thing. This is not affected by any of the properties mentioned above:

![frame-and-sheet-name](https://user-images.githubusercontent.com/150189/98588277-b7a9dc00-22cb-11eb-9fea-0c64764f02d1.png)

Maybe the "non-pretty name" is used here? Do we map that to some X concept? I can't see anything with the X introspection tools. Our code contains `(xlib:set-wm-class window instance-name class-name)` so maybe that's the source.

> do we want the icon-mask? it may be redundant with the transparency, so I think not.

Yes, the specification explicitly says ARGB, so it is not a problem for X. Should a backend need a separate mask, it could still extract the alpha channel of the ARGB image pattern.

Reply:
> after a cursory glance at clim tos, they seem to represent the icon as a lambda list: `(&key (icon-name (frame-pretty-name frame)) pixmap clipping-mask)` and destructure it in the backend.

I think it would be better to extend `:pretty-name` instead since the icon is not tied to the frame being iconified nowadays and similarly, a long name and a short name may be useful outside of iconified frames. The syntax could be like `(:pretty-name PRETTY-NAME :icon-name ICON-NAME)` or `(:pretty-name (:default PRETTY-NAME :short SHORT-NAME))` or similar.

Reply:
closing (after discussion on irc).